### DISCUSSION TITLE
NB BUILD FAILED (gradle module)

### DISCUSSION THREAD

**Discussion by sai-vivekanand** [apache/netbeans#6612] [General]
in the local, cloned the repo, executed 'ant build' and it failed at :
BUILD FAILED
/Users/vsvr/openSource/netbeans/nbbuild/build.xml:635: The following error occurred while executing this line:
/Users/vsvr/openSource/netbeans/nbbuild/build.xml:630: The following error occurred while executing this line:
/Users/vsvr/openSource/netbeans/nbbuild/build.xml:677: The following error occurred while executing this line:
/Users/vsvr/openSource/netbeans/extide/gradle/build.xml:61: Java returned: 1

anyone knows how to resolve this issue? I looked through gradle/build.xml at 61st line and it's:
<java fork="true" dir="${tooling}" classpath="${tooling}/gradle/wrapper/gradle-wrapper.jar" classname="org.gradle.wrapper.GradleWrapperMain" failonerror="true">
            <sysproperty key="org.gradle.appname" value="Gradle"/>
            <arg line="--offline"/>
            <arg line="clean build -x check"/>
        </java>
        <copy file="${tooling}/build/libs/${tooling}.jar" todir="build/tooling" overwrite="true"/>
        <copy file="${tooling}/src/main/resources/nb-tooling.gradle" todir="build/tooling" overwrite="true"/>
    </target>

and nbbuild/build.xml lines from 630 are:
<macrodef name="run-depend-build">
    <attribute name="clusters-to-build"/>
    <sequential>
      <repeat target="build-one-cluster" name="one.cluster.name" values="@{clusters-to-build}" /> 
    </sequential>
  </macrodef>

  <target name="build-clusters" depends="init">
     <run-depend-build clusters-to-build="${nb.clusters.list}"/> 
  </target>

please help me if anyone resolved similar issues and what should be change to resolve this problem?

<img width="1440" alt="build failed_netbeans" src="https://github.com/apache/netbeans/assets/141432630/239f5efd-3a87-427a-b31c-44c216ca9e77">

**Comment by neilcsmith-net** [comment]
I assume you're trying to build with Java 21?  Either switch to Java 11 or 17, or read the notes in the README about using an alternative JDK to build the Gradle support.

**Comment by mbien** [comment]
special setup is now no longer needed https://github.com/apache/netbeans/pull/6816
